pkg/db/db.go

```diff
@@ -217,6 +221,32 @@ func (c *Client) initOCIArtifact(opt types.RegistryOptions) (*oci.Artifact, erro
 	return art, nil
 }
 
+func (c *Client) downloadDB(ctx context.Context, opt types.RegistryOptions, dst string) error {
+	downloadOpt := oci.DownloadOption{MediaType: dbMediaType}
+	if c.artifact != nil {
+		return c.artifact.Download(ctx, dst, downloadOpt)
+	}
```

Comment:
Do we need this?
We do same in initOCIArtifact:
https://github.com/aquasecurity/trivy/blob/95df470087ce12b6554403058329b4c89082caab/pkg/db/db.go#L202-L204
+
https://github.com/aquasecurity/trivy/blob/95df470087ce12b6554403058329b4c89082caab/pkg/db/db.go#L237-L239

Reply:
If we already have one artifact initialized with a repository via `WithOCIArtifact`, there is no point in looping through repositories.

Reply:
Maybe I didn't make it clear.

You have duplicate code.
Can we keep `c.artifact != nil` check and `a.Download` function in just one place?

https://github.com/aquasecurity/trivy/blob/95df470087ce12b6554403058329b4c89082caab/pkg/db/db.go#L226-L228
https://github.com/aquasecurity/trivy/blob/95df470087ce12b6554403058329b4c89082caab/pkg/db/db.go#L202-L204

and 

https://github.com/aquasecurity/trivy/blob/95df470087ce12b6554403058329b4c89082caab/pkg/db/db.go#L227
https://github.com/aquasecurity/trivy/blob/95df470087ce12b6554403058329b4c89082caab/pkg/db/db.go#L237

Reply:
@DmitriyLewen If the artifact already exists (manually created), then we don't need to create it [here](https://github.com/aquasecurity/trivy/pull/7605/files#diff-5da3684806835246d262230050593f460b12b6c0e3966df174e6061be0e9e575R230-R234) and we can just download it instead of trying to load it from possible repositories.

Reply:
@DmitriyLewen Done https://github.com/aquasecurity/trivy/pull/7605/commits/b08004ab47c824c5de7cb04011ccfcdfd2e20365